### PR TITLE
chore (condo): DOMA-10452 improve error messages for meters

### PR DESCRIPTION
**Pull request by AleX83Xpert**

### CODE REVIEW

File: apps/condo/lang/en/en.json [hunk 1]
```diff
@@ -3168,6 +3168,8 @@
   "acquiringReceipt.invoice.row.amount": "Amount",
   "api.acquiring.multiPayment.RECEIPTS_WITH_INVOICES_FORBIDDEN": "Receipts and invoices are forbidden to be together",
   "api.acquiring.payment.PAYMENT_AMOUNT_LESS_THAN_MINIMUM": "Payment amount must be at least {minimumPaymentAmount}",
-  "api.acquiring.payment.PAYMENT_AMOUNT_GREATER_THAN_MAXIMUM": "Payment amount must not exceed {maximumPaymentAmount}"
+  "api.acquiring.payment.PAYMENT_AMOUNT_GREATER_THAN_MAXIMUM": "Payment amount must not exceed {maximumPaymentAmount}",
+  "ISO_DATE_FORMAT": "YYYY-MM-DD",
+  "EUROPEAN_DATE_FORMAT": "DD.MM.YYYY"
```
Comment (Alllex202) [inline apps/condo/lang/en/en.json:3173]: keys has an invalid format

Comment (AleX83Xpert) [reply apps/condo/lang/en/en.json]: fixed

File: apps/condo/domains/meter/utils/taskSchema/AbstractMetersImporter.js [hunk 2]
```diff
@@ -178,7 +178,14 @@ class AbstractMetersImporter {
...
                             const messageForUser = get(mutationError, ['extensions', 'messageForUser'])
-                            const rowErrors = [messageForUser || internalMessage || message]
+
+                            const originalError = get(mutationError, ['originalError', 'errors', 0, 'originalError', 'errors', 0])
+                            const originalMessage = get(originalError, ['message'])
+                            const originalInternalMessage = get(originalError, ['extensions', 'message'])
+                            const originalMessageForUser = get(originalError, ['extensions', 'messageForUser'])
+
+                            // We need to show as understandable error as possible
+                            const rowErrors = [originalMessageForUser || originalInternalMessage || originalMessage || messageForUser || internalMessage || message]
```
Comment (sitozzz) [inline apps/condo/domains/meter/utils/taskSchema/AbstractMetersImporter.js:188]: Do we really need to pass internalMessage for user? I suppose it'll be quite unclear)

Comment (sitozzz) [reply apps/condo/domains/meter/utils/taskSchema/AbstractMetersImporter.js:188]: Don't we have errors by default?

Comment (AleX83Xpert) [reply apps/condo/domains/meter/utils/taskSchema/AbstractMetersImporter.js:188]: Yes. Most errors happen when custom mutation try to create/update a model. For example, custom mutation trying to create a meter and the Meter model throws an error. In this case a have a understandable error from model wrapped by custom mutation error. So, I show the model's error if exists. I've added a test for such case.

Comment (AleX83Xpert) [reply apps/condo/domains/meter/utils/taskSchema/AbstractMetersImporter.js:188]: > Don't we have errors by default?

This errors lift up into the error file, so it's mandatory to has an error from model.

**Review by sitozzz — APPROVED** [review]

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/domains/meter/tasks/importMeters.spec.js [hunk 3]
```diff
@@ -395,4 +442,54 @@ describe('importMeters', () => {
         })
         expect(readings).toHaveLength(validLines)
     })
+
+    test('error file contains ru-error about meter number exists on another account number', async () => {
+        const locale = 'ru'
+
```
[the remark this answers is not in the document]
Comment (AleX83Xpert) [reply apps/condo/domains/meter/tasks/importMeters.spec.js:448]: ¯\_(ツ)_/¯
Looks like this one is anough.

File: apps/condo/domains/meter/utils/taskSchema/AbstractMetersImporter.js [hunk 4]
```diff
@@ -178,7 +178,14 @@ class AbstractMetersImporter {
                             const message = get(mutationError, ['message'])
                             const internalMessage = get(mutationError, ['extensions', 'message'])
                             const messageForUser = get(mutationError, ['extensions', 'messageForUser'])
-                            const rowErrors = [messageForUser || internalMessage || message]
+
+                            const originalError = get(mutationError, ['originalError', 'errors', 0, 'originalError', 'errors', 0])
```
Comment (pahaz) [inline apps/condo/domains/meter/utils/taskSchema/AbstractMetersImporter.js:182]: It doesn't work in prod. You need to write a test for that.

**Pull request merged**